lib/tiled/tiled_object.rb

```diff
@@ -4,13 +4,15 @@ class TiledObject
     include Tiled::WithAttributes
 
     attr_reader :map
-    attributes :id, :x, :y, :width, :height, :shape, :points
+    attributes :id, :gid, :x, :y, :width, :height, :type, :points
 
     def initialize(map, attrs, children)
       @map = map
-      attributes.add(id: attrs['id'], shape: children.any? ? children.first[:name].to_sym : :rectangle)
+      attributes.add(id: attrs['id'], gid: attrs['gid']&.to_i)
 
-      if shape == :polygon
+      attributes.add(type: children.any? ? children.first[:name].to_sym : gid ? :tile : :rectangle)
```

Comment:
`type` is an object attribute in TMX format, so we should not replace it with our value. Maybe `object_type`?

Very hard to parse this expression, I would prefer something like this:
```ruby
def initialize(map, attrs, children)
  # ...
  attributes.add(object_type: detect_type(attrs))
  # ...
end

def detect_type(attrs, children)
  if children.any?
    children.first[:name].to_sym
  elsif attrs['gid']
    :tile
  else
    :rectangle
  end
end
```